### PR TITLE
Use timezone via settings to get PREFER_DATES_FROM result

### LINKED ISSUE #1092: Returned datetime skips a day when timezone is supplied via settings instead of date string
#1002 promised to fix #403 and has been merged to master. As I discovered now it is only a partial fix.

Recall the issue was the dateparser would skip ahead a day if the time had already passed in UTC when preferring dates from the future. For example parsing 6pm EST (11pm UTC) at 10pm UTC would return _tomorrow_ 11pm, instead of _today_ 11pm UTC.

#1002 addresses this by passing the time zone so the correct relative base can be computed. What I didn't realize at the time is that ptz is only the time zone popped off the string. So, if the timezone is supplied via settings instead, nothing is passed along and the old bug from #403 reappears.

I might submit a pull request in the future but I wanted to report it already.

**Versions:**

- Ubuntu 22.04.1 LTS
- Python 3.10.6
- dateparser (1.1.3)
  - python-dateutil (2.8.2)
  - pytz (2022.6)
  - regex (2022.3.2)
  - tzlocal (4.2)

**Minimum working example:**
```
import dateparser
import datetime

working_settings={
        'PREFER_DATES_FROM': 'future',
        'TO_TIMEZONE': 'etc/utc',
        'RETURN_AS_TIMEZONE_AWARE': False,
        'RELATIVE_BASE': datetime.datetime(2022, 11, 6, 22, 0)
        }

broken_settings = working_settings
broken_settings['TIMEZONE'] = 'america/new_york'

correct_time = dateparser.parse('6pm EST', settings=working_settings)
print(correct_time) # 2022-11-06 23:00:00

wrong_time = dateparser.parse('6pm', settings=broken_settings)
print(wrong_time) # 2022-11-07 23:00:00
```

### PR DESCRIPTION
**Pull request by gutsytechster**
This fixes #1092

### CODE REVIEW

File: tests/test_date_parser.py [hunk 1]
```diff
@@ -869,27 +869,20 @@ def test_parsing_strings_containing_only_separator_tokens(self, date_string, exp
         self.then_date_obj_exactly_is(expected)
 
     @parameterized.expand([
-        param('4pm EDT', datetime(2021, 10, 19, 20, 0)),
```
[the remark this answers is not in the document]
Comment (gutsytechster) [reply tests/test_date_parser.py:872]: I actually wanted to keep the similar timezone values provided through the `TIMEZONE` settings and within the date string which would enforce the idea that the logic is applicable to the inputs regardless of how the timezone value is provided. 

The `EDT` is parsed through the date string. However, it isn't a valid value for the `TIMEZONE` as it is not parsable via `pytz.timezone`.

[the remark this answers is not in the document]
Comment (Gallaecio) [reply tests/test_date_parser.py:872]: So with this change we stop supporting `EDT`. I wonder if we should consider an alternative to EDT, e.g. [zoneinfo](https://docs.python.org/3/library/zoneinfo.html) (with its backport for Python 3.8 and earlier).

[the remark this answers is not in the document]
Comment (gutsytechster) [reply tests/test_date_parser.py:872]: When `EDT` is passed along with the date string i.e. ` 6 PM EDT`, the time is correctly parsed. However, when it is passed through the `TIMEZONE` settings, the `pytz.timezone` cannot pick this. It's an unknown timezone for it. Therefore, it considers the timezone as None in such a case.

Now I wonder how does timezone from the date string is being picked. I may find some method within the lib which does the same.

[the remark this answers is not in the document]
Comment (gutsytechster) [reply tests/test_date_parser.py:872]: I used the inbuilt method `get_timezone_from_tz_string` which also processes timezone values unknown to `pytz.timezone` through an internally maintained list.

The previous test cases work now.

**Comment by Gallaecio** [comment]
I have run the minimum working example from the [original report](https://github.com/scrapinghub/dateparser/issues/1092), and the output is the same before and after this change.

**Comment by Gallaecio** [comment]
Today I Learn `pip install git+…` does not actually overwrite the installed package if they both have the same version in `setup.py` :sweat:

**Pull request merged**